### PR TITLE
Add custom backend for google-oauth

### PR DESCRIPTION
**Pull request by mumarkhan999**
A `custom google_oauth` has been added in this PR to replace the `google_oauth` of `social_core_auth`.

### CODE REVIEW

File: common/djangoapps/third_party_auth/custom_backends.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        name, given_name, family_name = (
+            response.get('name', ''),
+            response.get('given_name', ''),
+            response.get('family_name', ''),
+        )
+
+        fullname, first_name, last_name = self.get_user_names(
+            name, given_name, family_name
+        )
+        return {'username': email.split('@', 1)[0],
```
Comment (asamolion) [inline common/djangoapps/third_party_auth/custom_backends.py]: Please use better formatting here.

File: common/djangoapps/third_party_auth/custom_backends.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                'email': email,
+                'fullname': fullname,
+                'first_name': first_name,
+                'last_name': last_name}
+
+    def user_data(self, access_token, *args, **kwargs):
+        """Return user data from Google API"""
+        return self.get_json(
+            'https://www.googleapis.com/oauth2/v3/userinfo',
```
Comment (asamolion) [inline common/djangoapps/third_party_auth/custom_backends.py]: Don't hard code this endpoint. Please use a proper settings variable.

**Comment by asamolion** [comment]
Has this been tested locally and on the dev server?

File: common/djangoapps/third_party_auth/custom_backends.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
+from social_core.backends.google import GoogleOAuth2
+
+
+class CustomGoogleOAuth(GoogleOAuth2):
+
+    def get_user_id(self, details, response):
+        """Use google email as unique id"""
+        if self.setting('USE_UNIQUE_USER_ID', False):
```
[the remark this answers is not in the document]
Comment (mumarkhan999) [reply common/djangoapps/third_party_auth/custom_backends.py]: Didn't search much about that. But in the new version of  `social_auth_core` they have changed the endpoint of `google+` and used some extra fields like `sub` I have done exactly the same thing.

File: common/djangoapps/third_party_auth/custom_backends.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
+from social_core.backends.google import GoogleOAuth2
+
+
+class CustomGoogleOAuth(GoogleOAuth2):
```
[the remark this answers is not in the document]
Comment (mumarkhan999) [reply common/djangoapps/third_party_auth/custom_backends.py]: done in the latest commit.

**Pull request merged**